turbomoleio/output/tests/test_files.py

```diff
@@ -56,18 +56,18 @@ def cls_dict_path(request, testdir):
     fpaths = []
     if tm_exec == "escf":
         output_clses.append(EscfOnlyOutput)
-        ref_dicts.append(loadfn(os.path.join(path, "ref_escf_output.json"), cls=None))
+        ref_dicts.append(loadfn(os.path.join(path, "ref_escf_output.json")).as_dict())
```

Comment:
I am not sure what is the advantage of this change. It adds a serialization+deserialization and if there is a bug in the as_dict it can be missed, since the references will also pass through the as_dict.

Reply:
Hmmm I think I did it so that old reference dict files would have the new default values of a new version of the object. Not sure if I am clear here... so let me take an example:

Gradient now has the periodicity, lattice_vectors and lattice_gradients arguments which respectively defaults to 0, None and None. The reference files for TM 7.3.1 have been dumped with the old version (which did not have these arguments). As I do not want to rerun the older TM versions, I serialize then deserialize the obejct to have the default values for the new arguments. Do you think we should instead do this in the assert_almost_equal method somehow ? Doing the way its done right now also allows to check that deserialization of the old objects (but maybe its also done elsewhere, I'm lost now :p).

What do you think ?

Reply:
I see the point now.

I can't find a good solution though: 

- Leaving it like this still seems a bit dangerous. 
- How to handle it in assert_almost_equal? It should somehow ignore the "default" values, but the information whether a value is not the default cannot be known there, since it does not have the reference to the object. (and even with the object checking this would require introspection on the object)
- rerun TM for all the past supported versions every time a new parsed value is added is a pain

What about making an if on `tm_version` and use the standard solution only for those that are currently most supported and use the one with `as_dict` for the older ones? Would it be bad?

Reply:
Let's use the bare dictionary as before. And if needed we do something later.